pkg/runtime/interfaces.go

```diff
@@ -20,29 +20,23 @@ import (
 	"io"
 )
 
-// ObjectScheme represents common conversions between formal external API versions
-// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
-// transform internal Go structs into serialized versions. There may be many valid
-// ObjectCodecs for each ObjectScheme.
-type ObjectScheme interface {
-	ObjectConvertor
-	ObjectTyper
-	ObjectCreater
-	ObjectCopier
-}
-
-// ObjectCodec represents the common mechanisms for converting to and from a particular
-// binary representation of an object.
-type ObjectCodec interface {
-	ObjectEncoder
+// Codec defines methods for serializing and deserializing API objects.
```

Comment:
Sorry - I changed the order of definitions in this file, because it was difficult to follow there.

So summarize the changes:
- moved ObjectScheme to fixture.go - it is used only there and I think we should remove it completely
- removed ObjectEncoder interface completely (merged with ObjectCodec)
- added some TODOs for methods/interfaces that I think should be removed
Please take a look if you agree with that.